backend/src/apiserver/server/visualization_server.go

```diff
@@ -115,13 +112,14 @@ func (s *VisualizationServer) getVisualizationServiceURL(request *go_client.Crea
 	return s.serviceURL
 }
 
-func isVisualizationServiceAlive(serviceURL string) bool {
+func isVisualizationServiceAlive(serviceURL string) (bool, error) {
 	resp, err := http.Get(serviceURL)
-	if err != nil {
-		glog.Error("Unable to verify visualization service is alive!", err)
-		return false
+	if resp != nil && resp.StatusCode != http.StatusOK {
+		return false, errors.New(fmt.Sprintf("Unable to verify visualization service is alive by sending request to %s and get %s !", serviceURL, resp.Status))
+	} else if resp == nil && err != nil {
+		return false, util.Wrap(err, "The visualization server host not exist!")
```

Comment:
nit: when there's an error here, it doesn't always mean visualization server host doesn't exist.
`util.Wrap` keeps internal err's message, so in this layer, we just need to tell "what we are trying to do, and it failed"

Therefore, I'd suggest
```
return false, util.Wrap(err, fmt.Sprintf("Unable to verify visualization service is alive by sending request to %s", serviceURL))
```

Reply:
You can read documentation in https://golang.org/pkg/net/http/#Client.Get, the error can be other reasons, not just `server host not exist`.

Reply:
Agree, we need to make `isVisualizationServiceAlive` generic, but make `TestGenerateVisualization_ServiceHostNotExistError` test restricted to ServiceNotExistError.